modules/markup/markdown/markdown_test.go

```diff
@@ -507,21 +507,29 @@ func TestMathBlock(t *testing.T) {
 			`\(a\) \(b\)`,
 			`<p><code class="language-math is-loading">a</code> <code class="language-math is-loading">b</code></p>` + nl,
 		},
+		{
+			`$a$.`,
+			`<p><code class="language-math is-loading">a</code>.</p>` + nl,
+		},
+		{
+			`.$a$`,
+			`<p>.$a$</p>` + nl,
+		},
 		{
 			`$a a$b b$`,
-			`<p>$a a<code class="language-math is-loading">b b</code></p>` + nl,
+			`<p>$a a$b b$</p>` + nl,
```

Comment:
Could there be a case for something like:

* `$a a\\$b b$` => $a a\\$b b$

?

In case the users would like to escape and use `$`

Reply:
The old code does what you described with `$a a\$b b$`, with only one backslash.


With the current code, the expression `$a a\\$b b$` generates the expression below. 
![image](https://github.com/go-gitea/gitea/assets/114936010/6de6a822-15df-4806-840c-5c0a30e6a337)


If the user wanted to get the output that you described he would need to use the following expression
`$a a \$ b b$`
![image](https://github.com/go-gitea/gitea/assets/114936010/def20342-2ede-4126-b1ac-10af7d317991)